extensions/simple-mcp-server/main.py

```diff
@@ -0,0 +1,147 @@
+from fastapi.templating import Jinja2Templates
+from mcp.server.fastmcp import FastMCP, Context
+from mcp.server.fastmcp.exceptions import ToolError
+from posit.connect.client import Client as ConnectClient
+import urllib
+
+# --- FastMCP Server Initialization ---
+mcp = FastMCP(
+    name="SimpleDataServer",
+    instructions="MCP server for dataset operations and Connect 'whoami' via FastAPI.",
+)
+
+# --- Datasets ---
+# Simple in-memory datasets for demonstration
+_datasets_store = {
+    "iris": lambda: pd.DataFrame(data=load_iris(as_frame=True).frame),
+    "sample_data": lambda: pd.DataFrame(
+        {"A": [1, 2, 3, 4, 5], "B": [5, 4, 3, 2, 1], "C": ["x", "y", "x", "z", "y"]}
+    ),
+}
+
+
+# --- MCP Tool Implementations ---
+@mcp.tool()
+def list_known_datasets() -> str:
+    """Lists available dataset names."""
+    return str(list(_datasets_store.keys()))
+
+
+@mcp.tool()
+def calculate_summary_statistics(dataset_name: str) -> str:
+    """
+    Calculates summary statistics for a specified dataset.
+    Returns the summary as a string or an error.
+    """
+    if dataset_name not in _datasets_store:
+        raise ToolError(f"Dataset '{dataset_name}' not found.")
+    try:
+        df = _datasets_store[dataset_name]()
+        summary = df.describe(include="all").to_string()
+        return summary
+    except Exception as e:
+        raise ToolError(f"Error processing dataset '{dataset_name}': {str(e)}")
+
+
+# This tool is unable to be used in conjunction with the Simple Shiny Chat app 
+# @mcp.tool()
+# async def connect_whoami(context: Context) -> str:
+#     """
+#     Calls the Posit Connect /me endpoint using an API key from the Authorization header.
+#     The Authorization header should be in the format: 'Key YOUR_API_KEY'.
+#     """
+
+#     # context.request is a starlette.requests.Request
+#     http_request = context.request_context.request
+#     if http_request is None:
+#         raise ToolError(
+#             "Request context not available. This tool requires an HTTP-based transport."
+#         )
+
+#     auth_header = http_request.headers.get("x-mcp-authorization")
+
+#     if not auth_header:
+#         raise ToolError("Authorization header is missing.")
+
+#     parts = auth_header.split()
+#     if len(parts) != 2 or parts[0].lower() != "key":
+#         raise ToolError(
+#             "Invalid Authorization header format. Expected 'Key YOUR_API_KEY'."
+#         )
+
+#     api_key = parts[1]
+
+#     try:
+#         connect_client = ConnectClient(api_key=api_key)
+#         return json.dumps(connect_client.me)
+#     except Exception as e:
+#         raise ToolError(f"Error calling Connect API: {str(e)}")
+
+
+@contextlib.asynccontextmanager
+async def lifespan(app: FastAPI):
+    async with contextlib.AsyncExitStack() as stack:
+        await stack.enter_async_context(mcp.session_manager.run())
+        yield
+
+
+app = FastAPI(title="Simple MCP Server with FastAPI", lifespan=lifespan)
```

Comment:
TIL that python's MCP let's you piggyback on top of Fast API!  That's wonderful... I had thought that we might have to add an additional content type for the python MCP SDK, so that's super nice that we get it for free!